clearwater-etcd/usr/share/clearwater/clearwater-etcd/scripts/save_etcd_config.py

```diff
@@ -0,0 +1,121 @@
+# Project Clearwater - IMS in the Cloud
+# Copyright (C) 2016 Metaswitch Networks Ltd
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version, along with the "Special Exception" for use of
+# the program along with SSL, set forth below. This program is distributed
+# in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details. You should have received a copy of the GNU General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+# The author can be reached by email at clearwater@metaswitch.com or by
+# post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+#
+# Special Exception
+# Metaswitch Networks Ltd  grants you permission to copy, modify,
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+import json
+import re
+import subprocess
+import sys
+
+
+# Gets the value from etcd for the given key
+def get_value(key):
+    stored_value = subprocess.check_output(["clearwater-etcdctl", "get", key])
+    return stored_value.strip()
+
+
+# Save the config
+def save_to_disk(data):
+    with open("/tmp/saved_etcd_config", "w+") as saved_config_file:
+        json.dump(data, saved_config_file)
+
+
+# Produces a dictionary like {"[node_ip]": "normal"} containing every cassandra
+# node in the local site
+def get_local_cassandra_nodes(site_name):
+    nodetool_process = subprocess.Popen("nodetool status", stdout=subprocess.PIPE, shell=True)
```

Comment:
This needs to use the signalling namespace (use /usr/share/clearwater/bin/run-in-signaling-namespace, but please check the path)